api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -944,17 +944,20 @@ message Cluster {
   // [#next-major-version: make this a list of typed extensions.]
   map<string, google.protobuf.Any> typed_extension_protocol_options = 36;
 
-  // If the DNS refresh rate is specified and the cluster type is either
+  google.protobuf.Duration dns_refresh_rate = 16
+      [(validate.rules).duration = {gt {nanos: 1000000}}];
+
+  // If the DNS jitter is specified the cluster type is either
   // :ref:`STRICT_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.STRICT_DNS>`,
-  // or :ref:`LOGICAL_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.LOGICAL_DNS>`,
-  // this value is used as the cluster’s DNS refresh
-  // rate. The value configured must be at least 1ms. If this setting is not specified, the
-  // value defaults to 5000ms. For cluster types other than
+  // or :ref:`LOGICAL_DNS<envoy_v3_api_enum_value_config.cluster.v3.Cluster.DiscoveryType.LOGICAL_DNS>`.
+  // DNS jitter causes the cluster to refresh DNS entries early by a round amount
+  // to avoid a stampede of DNS requests. This value sets the upper bound for the random amount (exclusive).
+  // Setting this value to 1ms will disable jitter.
```

Comment:
It is not clear to me why 1ms has a special value?
I think the behavior here should be only apply jitter if it is configured. This will also be backward compatible.